fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateInnerJoinByEmptyRelation.java

```diff
@@ -0,0 +1,61 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite.logical;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.rules.rewrite.OneRewriteRuleFactory;
+import org.apache.doris.nereids.trees.plans.GroupPlan;
+import org.apache.doris.nereids.trees.plans.JoinType;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalEmptyRelation;
+import org.apache.doris.nereids.trees.plans.logical.LogicalJoin;
+
+/**
+ * before:
+ *  inner-join
+ *     /     \
+ *   Node    EmptyRelation
+ *
+ * after:
+ *   EmptyRelation
+ */
+
+public class EliminateInnerJoinByEmptyRelation extends OneRewriteRuleFactory {
```

Comment:
I suggest rename `PropagateEmpty`.

And, Put all type empty pattern in this rule.

Reply:
good suggestion.
but 
1. it will make PropagateEmpty.java too huge.
2. empty relation is not always propagated. for example, anti-join has different result.